Src/IronPython/Runtime/PythonFileManager.cs

```diff
@@ -28,10 +29,19 @@ internal sealed class StreamBox {
         private int _id = -1;
         private Stream _readStream;
         private Stream _writeStream;
+        private readonly bool _writeNeedsFlush;
 
         public StreamBox(Stream readStream, Stream writeStream) {
             _readStream = readStream;
             _writeStream = writeStream;
+
+            // a hack but it does improve perfomance a lot if flushing is not needed
+            _writeNeedsFlush = writeStream switch {
+                FileStream => true,         // FileStream is buffered by default, used on Windows and mostly Mono
```

Comment:
I've just checked the source code of .NET Framework 4.8 and it also performs unbuffered writes when buffer size is 1. I suspect it is the same on 4.6.2. On Mono, however, buffer bypass only happens if the data length is > buffer size, meaning that single-byte writes still land in the 1-byte buffer, not on the disk. We could then optimize flushing by only flushing if IsMono and data length is 1.

But `FileIO.OpenFile` is not the only way of creating `FileIO`. One can use `os.open` to get a file descriptor and then open a file. Currently, `PythonNT.open`, when creating `FileStream`, uses buffer of 4K (`DefaultBufferSize`), which is defined since the initial commit 11 years ago. I will change it to 1 to get the consistent unbuffered behaviour, unless you have counter arguments.

Reply:
I think we can also open with a buffer size of 0 if that would help with Mono...

Reply:
I don't think so. Mono, .NET, and .NET Framework all throw for buffer size <= 0.

Reply:
Hmm, in the [`FileStreamOptions.BufferSize`](https://learn.microsoft.com/en-us/dotnet/api/system.io.filestreamoptions.buffersize#system-io-filestreamoptions-buffersize) docs it says `0` or `1` to disable buffering. Although the class is new (.NET 6) I was assuming it'd simply pass along the value. Anyway, it looks like .NET Framework does indeed throw for <= 0 so I guess it doesn't help.

Reply:
You are right, .NET doesn't throw with 0, I was speaking from memory, must have gotten confused with another framework. Or maybe it was throwing in the past (older framework versions).

Anyway, the issue is with Mono, and it does [throw at 0](https://github.com/mono/mono/blob/0f53e9e151d92944cacab3e24ac359410c606df6/mcs/class/corlib/System.IO/FileStream.cs#L167-L168), and it does [buffer at 1](https://github.com/mono/mono/blob/0f53e9e151d92944cacab3e24ac359410c606df6/mcs/class/corlib/System.IO/FileStream.cs#L617).